Major Restructure and Re-Formatting

NickVendel opened the pull request:
The point of this update is to apply one singular standard to all files within this repository, and to fix minor issues or bugs.

**Breaking Changes**
Adventurer Background: 
• Changed ID of the Background Feature to fit name of the feature. Would automatically reapply upon loading the character.
• Updated names of the selects. Users will have to re-select Skill Proficiencies granted by this Background.
Cantrip Caster Feat: 
• Updated select name. Users will have to re-select chosen for this Feat Class and Cantrip.
Homunculus Race: 
• Changed how ASI works, so it no longer needs ASI Racial Traits. Users will have to re-select Ability Score Increases granted by this Race in the Ability Score menu.
Spectral Scythe Spell: 
• Changed support from "Bard" to "Cleric". Bard characters that used this spell will lose the spell.

NickVendel committed:
Major restructure and re-formatting

• dndwiki.index: restructured to be in alphabetical order. Updated names and links of the files.
• Adventurer Background: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, changed ID of the Background Feature (minor breaking change), added short description for the selection, updated and added links to reference specific version of the article, updated names of the selects (breakning change).
• Apothecary Background: changed info about the file creation to fit new standard, changed info description to include authors/contributors, fixed spaces vs tabs, added short description for the selection, updated and added links to reference specific version of the article, fixed wrong description and skill proficiency, fixed Language selection to be limited to Standard and Exotic languages, not Secret ones, removed comments.
• Bounty Hunter Background: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, added short description for the selection, updated and added links to reference specific version of the article, added alt. name to the "Variant: Vigilante" Background Variant.
• Islander Background: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, removed text under Origin that doesn't appear anywhere in the article, added short description for the selection, updated and added links to reference specific version of the article.
• Prospector Background: changed info about the file creation to fit new standard, changed info description to include authors/contributors, fixed spaces vs tabs, removed comments, added short description for the selection, updated and added links to reference specific version of the article, removed tool from the file and added it to a separate file.
• Witch Hunter Background: renamed file and obsoleted old one, added info about the file creation to fit new standard, changed info description to include authors/contributors, fixed spaces vs tabs, added short description for the selection, updated and added links to reference specific version of the article, fixed Language selection to only allow Exotic languages.
• Necromancer Class: added info about the file creation to fit new standard, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added table to Necromancer, changed <h5> to <h3> for "Variant Feature: Grim Harvest" text.
• Weapon Master Class: added info about the file creation to fit new standard, changed info description to include authors/contributors, fixed description to better represent article, combined <span> lines with <br /> lines, added table for Weapon Master, updated short description, fixed wrong saving throw proficiency, removed comments.
• Circle of the Claw Subclass: added info about the file creation to fit new standard, changed info description to include authors/contributors, updated and added links to reference specific version of the article, updated sheet descriptions, removed pointless tags, added <div> references.
• Arcane Origin Subclass: added info about the file creation to fit new standard, changed info description to include authors/contributors, fixed spaces vs tabs.
• Feats: added info about the element creation to fit new standard, fixed spaces vs tabs, updated and added links to reference specific version of the article, updated sheet descriptions, removed pointless tags, updated select name for Cantrip Caster (breaking change), removed sub-elements for the selection from compendium.
• Items - Tools: created the file for the Prospector's kit and its proficiency, added info about the element creation to fit new standard, changed description of the Proficiency, updated and added links to reference specific version of the article.
• Items - Weapons: added info about the element creation to fit new standard, updated and added links to reference specific version of the article, updated sheet descriptions.
• Concubus Race: obsoleted file deleted from the repository.
• Dryad Race: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added TCOE rules for changing ASI and Languages, changed sheet descriptions, added empty line to the end of the file.
• Homunculus Race: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article,  updated descriptions to have spells in italics, added TCOE rules for changing ASI and Languages, changed how ASI works, so it no longer needs ASI Racial Traits (breaking change).
• Sprite Race: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added quote that was missing from main description, updated descriptions to have spells in italics, changed Trait "Clan" to "Subrace" to better represent the article, added TCOE rules for changing ASI and Languages, added div references, changed sheet descriptions and added action and usage properties, added Grants for "count as one size larger", slightly modified Tiny Fighter rules to have something similar to Tortle's Natural Armor.
• Succubus Race: added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added TCOE rules for changing ASI and Languages.
• Firebeard Dwarf Subrace: renamed the file and obsoleted the old one, added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added TCOE rules for changing ASI and Languages, fixed main description of the Racial Traits, removed sheet description of "Fire Resistance" Trait and added a div reference to "Firestrike" Trait.
• Rune Dwarf Subrace: renamed the file and obsoleted the old one, added info about the file creation, changed info description to include authors/contributors, fixed spaces vs tabs, updated and added links to reference specific version of the article, added TCOE rules for changing ASI and Languages, fixed main description of the Racial Traits.
• Source: renamed the file and obsoleted the old one, added short description to the source.
• Spells: added info about the element creation to fit new standard, fixed spaces vs tabs, updated and added links to reference specific version of the article, updated structure of the setters in order they actually displayed, changed support from "Bard" to "Cleric" in Spectral Scythe (potential breaking change).

AngryGuineaPig commented on dndwiki/races/race-concubus.xml:
```diff
@@ -1,98 +0,0 @@
-<?xml version="1.0" encoding="utf-8" ?>
```
is there a reason we just straight up deleted concubus?

NickVendel replied:
file was obsoleted before, because it's a copy of Succubus just gender-neutral.

AngryGuineaPig replied:
ah that makes sense

AngryGuineaPig commented:
Additionally for fire beard dwarf, rune dwarf and sources.xml I understand why they weren't currently deleted but since it's a major restructure would it not be possible to delete them and then ask people to re-download the entire index so we don't have obsolete files?

MeriadarGoblin commented on dndwiki/classes/class-necromancer.xml:
```diff
@@ -1,565 +1,611 @@
+		<description>
+			<p>Starting at 1st level, your touch can siphon the life force of others to bolster your own. As an action, you can make a melee spell attack against a living creature, dealing necrotic damage equal to 1d8 + your Charisma modifier on a hit. You gain temporary hit points equal to the amount of necrotic damage dealt. If this feature kills the creature, you gain twice as many temporary hit points ...
+		</description>
+		<sheet>
 			<description>{{life tap:count}}/long rest. As an action, you can make a melee spell attack against a living creature, dealing necrotic damage equal to 1d8 + your Charisma modifier on a hit. You gain temporary hit points equal to the amount of necrotic damage dealt. If this feature kills the creature, you gain twice as many temporary hit points from using this feature.</description>
 			<description level="5">{{life tap:count}}/long rest. As an action, you can make a melee spell attack against a living creature, dealing necrotic damage equal to 2d8 + your Charisma modifier on a hit. You gain temporary hit points equal to the amount of necrotic damage dealt. If this feature kills the creature, you gain twice as many temporary hit points from using this feature. </description>
 			<description level="11">{{life tap:count}}/long rest. As an action, you can make a melee spell attack against a living creature, dealing necrotic damage equal to 3d8 + your Charisma modifier on a hit. You gain temporary hit points equal to the amount of necrotic damage dealt. If this feature kills the creature, you gain twice as many temporary hit points from using this feature.</description>
 			<description level="17">{{life tap:count}}/long rest. As an action, you can make a melee spell attack against a living creature, dealing necrotic damage equal to 4d8 + your Charisma modifier on a hit. You gain temporary hit points equal to the amount of necrotic damage dealt. If this feature kills the creature, you gain twice as many temporary hit points from using this feature.</description>
```
<sheet> can be changed to <sheet action="Action" usage="Long Rest" /> as this class was made before this can be allowed.

MeriadarGoblin replied:
Sheet can be changed to allow action="Action" and usage="Long Rest" as they were not introduced at time of creation.

NickVendel commented:
> Additionally for fire beard dwarf, rune dwarf and sources.xml I understand why they weren't currently deleted but since it's a major restructure would it not be possible to delete them and then ask people to re-download the entire index so we don't have obsolete files?

I prefer obsoleting to "ask people", we don't have a way to communicate with most of those people, and even if we had, i don't really want to deal with people not understanding what to delete, etc. It's less intrusive to do it via obsoleting.

MeriadarGoblin commented on dndwiki/classes/class-necromancer.xml:
```diff
@@ -1,565 +1,611 @@
+			<p>Beginning at 7th level, your pursuit of knowledge has led to uncover the secrets to mastering undead of all sizes. You can now raise any corpse that is Huge or smaller of a monstrosity, humanoid, or beast with challenge rating 3 or lower can be animated. It keeps its original abilities and attacks, but loses any resistances, immunities, legendary actions, senses it had and gains the immunit ...
+			<p class="indent">At level 11, you may animate a monstrosity, humanoid or beast with a challenge rating of 6 or lower.</p>
+			<p class="indent">At level 15, you may animate a monstrosity, humanoid or beast with a challenge rating of 9 or lower. In addition, a major undead now lasts until its hit points are reduced to 0.</p>
+		</description>
```
Could use a div="reference" at the end of the description to display Animate Dead.

MeriadarGoblin commented on dndwiki/classes/class-necromancer.xml:
```diff
@@ -1,565 +1,611 @@
+
+	<element name="Ritualistic Unburial" type="Class Feature" source="D&amp;D Wiki" id="ID_WIKI_CLASS_FEATURE_NECROMANCER_RITUALISTIC_UNBURIAL">
+		<description>
+			<p>At level 14th level, the repetitive nature of raising your undead has become like a ritual for you. You may cast <i>animate dead</i> as a ritual.</p>
```
Could also use a div="reference" for Animate Dead however not as necessary

AngryGuineaPig commented:
> > Additionally for fire beard dwarf, rune dwarf and sources.xml I understand why they weren't currently deleted but since it's a major restructure would it not be possible to delete them and then ask people to re-download the entire index so we don't have obsolete files?
> 
> I prefer obsoleting to "ask people", we don't have a way to communicate with most of those people, and even if we had, i don't really want to deal with people not understanding what to delete, etc. It's less intrusive to do it via obsoleting.

yeah that's fair we can leave them alone

MeriadarGoblin commented on dndwiki/classes/druid-claw.xml:
```diff
@@ -19,6 +21,9 @@
+		<setters>
+			<set name="sourceUrl">https://www.dandwiki.com/w/index.php?title=Circle_of_the_Claw_(5e_Subclass)&amp;oldid=1156015</set>
+		</setters>
 		<sheet>
```
Is it necessary to have this being displayed on the sheet when it does nothing?

MeriadarGoblin replied:
I selected the wrong thing, I meant for the sheet node below that hehe

NickVendel replied:
not sure what you mean

MeriadarGoblin replied:
Is it necessary to have the description of the class to be on the sheet? I'm recommending it should be `<sheet display="false">`. Lines 27 to 29 are the code in question.

MeriadarGoblin commented on dndwiki/races/race-dryad.xml:
```diff
@@ -1,96 +1,118 @@
+			<set name="sourceUrl">https://www.dandwiki.com/w/index.php?title=Dryad_(5e_Race)&amp;oldid=954506#Dryad_Traits</set>
+		</setters>
+		<sheet usage="1/Turn">
+			<description>You can use 10 feet of your movement to step magically into one living tree within your reach and emerge from a second living tree within 60 feet of the first tree, appearing in an unoccupied space within 5 feet of the second tree.</description>
```
"Both trees must be Large or bigger." will be a nice add on for clarification for players.

NickVendel committed:
suggested changes

Co-Authored-By: MeriadarGoblin <99753019+MeriadarGoblin@users.noreply.github.com>

NickVendel committed:
additional changes

Weapon Master:
• fixed sourceUrl links
• shortened sheet descriptions, and added action property to Natural Leader feature's sheet description.
• removed pointless sheet descriptions.
Minor changes to Claw Druid and Arcane Sorcerer

MeriadarGoblin reviewed (changes requested):
All looks good except for a fix for dryad claw sheet description for the archetype.

NickVendel committed:
removed sheet description for Archetype elements
